splash/qtrender_lua.py

```diff
@@ -558,8 +568,12 @@ def set_result_header(self, name, value):
             raise ScriptError("splash:set_result_header() arguments must be strings")
 
         try:
-            name = name.decode('utf-8').encode('ascii')
-            value = value.decode('utf-8').encode('ascii')
+            try:
+                name = name.decode('utf-8').encode('ascii')
+                value = value.decode('utf-8').encode('ascii')
+            except AttributeError:
```

Comment:
Why is it necessary? I.e. why can name and value be of different types, can't they have a single type both in Python 3.x and 2.x?